Reworking api

psteinb opened the pull request:
@royerloic I changed the sqy api that is accessible from the C side (aka java downstream); I added the option to specify the number of threads when compressing/uncompressing. 

Please review the API if you can. If not, let me know if I can merge.

Commit:
removed obsolete C API bits

Commit:
added number of threads option

royerloic commented:
I like it! perfect!

Sent from my iPhone

> On 8. Sep 2017, at 16:18, Peter Steinbach <notifications@github.com> wrote:
> 
> @royerloic I changed the sqy api that is accessible from the C side (aka java downstream); I added the option to specify the number of threads when compressing/uncompressing.
> 
> Please review the API if you can. If not, let me know if I can merge.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/sqeazy/sqeazy/pull/45
> 
> Commit Summary
> 
> removed obsolete C API bits
> added number of threads option
> File Changes
> 
> M src/cpp/inc/sqeazy.h (342)
> M src/cpp/src/sqeazy.cpp (338)
> Patch Links:
> 
> https://github.com/sqeazy/sqeazy/pull/45.patch
> https://github.com/sqeazy/sqeazy/pull/45.diff
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Commit:
removed tests for old C API for now that probed the bits of API which
are to be removed, added comment about replacing their contents with
calls to SQY_PipelineEncode_*

Commit:
run unit tests in travis CI not just build

Commit:
removed old LZ4 calls

Commit:
fixed type of pipeline name to be c compatible

Commit:
upped pipeline API for encode/decode step

Commit:
added ignore file details

Commit:
more ignore statemetns
